src/components/slider/slider.spec.ts

```diff
@@ -0,0 +1,440 @@
+      return gestureConfig;
+    }})
+  ]);
+
+  beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
+    builder = tcb;
+  }));
+
+  describe('standard slider', () => {
+    let fixture: ComponentFixture<StandardSlider>;
+    let sliderDebugElement: DebugElement;
+    let sliderNativeElement: HTMLElement;
+    let sliderInstance: MdSlider;
+    let testComponent: StandardSlider;
+
+    beforeEach(async(() => {
+      builder.createAsync(StandardSlider).then(f => {
+        fixture = f;
+        fixture.detectChanges();
+
+        testComponent = fixture.debugElement.componentInstance;
+
+        sliderDebugElement = fixture.debugElement.query(By.directive(MdSlider));
+        sliderNativeElement = sliderDebugElement.nativeElement;
+        sliderInstance = sliderDebugElement.injector.get(MdSlider);
+      });
+    }));
+
+    it('should set the default values', () => {
+      expect(sliderInstance.value).toBe(0);
+      expect(sliderInstance.min).toBe(0);
+      expect(sliderInstance.max).toBe(100);
+    });
+
+    it('should update the value on a click', () => {
+      expect(sliderInstance.value).toBe(0);
+      dispatchClickEvent(sliderNativeElement, 0.5);
+      // 50% is the same as the value 50 in this case.
+      expect(sliderInstance.value).toBe(50);
+    });
+
+    it('should update the value on a drag', () => {
+      expect(sliderInstance.value).toBe(0);
+      dispatchDragEvent(sliderNativeElement, 0, 0.5, gestureConfig);
+      expect(sliderInstance.value).toBe(50);
+    });
+
+    it('should set the value as min when dragging before the track', () => {
+      expect(sliderInstance.value).toBe(0);
+      dispatchDragEvent(sliderNativeElement, 0, -1.5, gestureConfig);
+      expect(sliderInstance.value).toBe(0);
+    });
+
+    it('should set the value as max when dragging past the track', () => {
+      expect(sliderInstance.value).toBe(0);
+      dispatchDragEvent(sliderNativeElement, 0, 1.5, gestureConfig);
+      expect(sliderInstance.value).toBe(100);
+    });
+
+    it('should update the track fill on click', () => {
+      let trackFillElement = sliderNativeElement.querySelector('.md-slider-track-fill');
+      let trackFillDimensions = trackFillElement.getBoundingClientRect();
+      let sliderDimensions =
+          sliderNativeElement.querySelector('.md-slider-track').getBoundingClientRect();
+
+      expect(trackFillDimensions.width).toBe(0);
+      dispatchClickEvent(sliderNativeElement, 0.5);
+
+      trackFillDimensions = trackFillElement.getBoundingClientRect();
+      expect(trackFillDimensions.width).toBe(sliderDimensions.width * 0.5);
+    });
+
+    it('should update the thumb position on click', () => {
+      let thumbElement = sliderNativeElement.querySelector('.md-slider-thumb-position');
+      let thumbDimensions = thumbElement.getBoundingClientRect();
+      let thumbWidth =
+          sliderNativeElement.querySelector('.md-slider-thumb').getBoundingClientRect().width;
```

Comment:
You query for the track and the thumb pretty frequently; would it make sense to just get them once in the `beforeEach`?